### PR TITLE
ROX-22101: call gogo proto.Equal through compatibility layer

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The goal is to remove direct calls to the gogo protobuf layer and have all of them happen through an abstraction layer in order to ease the migration to a new protobuf library.

This PR moves the direct message comparisons from the gogo protobuf package to the protocompat abstraction layer.

The change is the application through the refactoring tool `eg` of the following template:

```
package refactor

import (
	"github.com/gogo/protobuf/proto"
	"github.com/stackrox/rox/pkg/protocompat"
)

func before(a proto.Message, b proto.Message) bool {
	return proto.Equal(a, b)
}

func after(a proto.Message, b proto.Message) bool {
	return protocompat.Equal(a, b)
}

```

PR stack:
- #9148
  - #9156 
  - #9145 
  - #9196 
  - #9198
  - #9201

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: tools/roxvet/analyzers/gogoprotofunctions/analyzer.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	allowedCallerPackages = []string{
+		"github.com/stackrox/rox/pkg/protocompat",
+		"github.com/stackrox/rox/pkg/protoconv",
+	}
+
+	monitoredFunctions = map[string]string{
+		// Key is the function name in gogo protobuf,
+		// value is the replacement function name in protocompat
+		"Equal": "Equal",
+	}
```
Comment (md2119) [inline tools/roxvet/analyzers/gogoprotofunctions/analyzer.go:33]: Is there a particular reason you have to check for individual function calls? You seem to remove all usages of `github.com/gogo/protobuf/types` from central and sensor through various PRs.

**Review by md2119 — COMMENTED** [review]
Nice work with the `roxvet`.

File: tools/roxvet/analyzers/gogoprotofunctions/analyzer.go (hunk 1 above)
Comment (rhybrillou) [reply tools/roxvet/analyzers/gogoprotofunctions/analyzer.go]: The point of this roxvet rule is to ensure that the gogoproto functions do not get called directly again in the codebase by code changes. The proto library change is a big chunk of work and concurrent changes may extend again the code impacts of the proto library change. We'd like to keep the changes contained.

**Review by janisz — APPROVED** [review]

**Review by mtodor — APPROVED** [review]

**Pull request merged**